htdocs/product/class/product.class.php

```diff
@@ -726,7 +747,7 @@ function update($id, $user, $notrigger=false, $action='update')
 		if (! $this->label) $this->label = 'MISSING LABEL';
 
 		// Clean parameters
-		$this->ref = dol_string_nospecial(trim($this->ref));
+		$this->ref = trim($this->ref);
```

Comment:
Why removing sanitizing ref ?
We know ref is also used to forge the PDF filename and we want/need a file name that is exactly what is ref. Removing this may break a lot of feature.

Reply:
then we should sanitize file creation
references can be anything, they depend on the provider and can contain slashes, especially if since this patch allows any kind of ref